send_stream.go

```diff
@@ -108,27 +109,46 @@ func (s *sendStream) Write(p []byte) (int, error) {
 		notifiedSender bool
 	)
 	for {
-		bytesWritten = len(p) - len(s.dataForWriting)
-		deadline := s.deadline
-		if !deadline.IsZero() {
-			if !time.Now().Before(deadline) {
-				s.dataForWriting = nil
-				return bytesWritten, errDeadline
+		var copied bool
```

Comment:
We should probably document the packing strategy somewhere in the code – even with the PR description it's quite hard to follow this.